### DISCUSSION TITLE
Unable to build project with avalonia from source to debug issue

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#12783] [Q&A] [answered]
Currently having an issue where Avalonia is MainViewModel into whatever ViewModel has a button linked to its own command

Tried to build Avalonia 11.0.4 from source and use it in my app to get more information about the issue but i'm unable to build my project as it keeps throwing errors about source generated stuff not existing, such as `InitializeComponent` & named Controls.

The following has been placed in the avalonia.xplat template

Here is part of the references from the core csproj :
```csproj
<ItemGroup>
    <ProjectReference Include="$(AVALONIA_ROOT)\packages\Avalonia\Avalonia.csproj" />
    <ProjectReference Include="$(AVALONIA_ROOT)\src\Avalonia.Themes.Fluent\Avalonia.Themes.Fluent.csproj" />
    <ProjectReference Include="$(AVALONIA_ROOT)\src\Avalonia.Fonts.Inter\Avalonia.Fonts.Inter.csproj" />
    <ProjectReference Include="$(AVALONIA_ROOT)\src\Avalonia.ReactiveUI\Avalonia.ReactiveUI.csproj" />
    <ProjectReference Include="$(AVALONIA_ROOT)\src\Avalonia.Diagnostics\Avalonia.Diagnostics.csproj" />
</ItemGroup>

<Import Project="$(AVALONIA_ROOT)\build\BuildTargets.targets" />
<Import Project="$(AVALONIA_ROOT)\build\SourceGenerators.props" />
<Import Project="$(AVALONIA_ROOT)\build\ReferenceCoreLibraries.props" />
```
As well as the .Desktop's csproj :
```csproj
<ItemGroup>
    <ProjectReference Include="$(AVALONIA_ROOT)\src\Avalonia.Desktop\Avalonia.Desktop.csproj" />
    <ProjectReference Include="$(AVALONIA_ROOT)\src\Avalonia.Diagnostics\Avalonia.Diagnostics.csproj" />
</ItemGroup>
```
running dotnet build result in the followings compile errors :
```
Views\MainView.axaml.cs(12,9): error CS0103: The name 'InitializeComponent' does not exist in the current context [Template\Template.csproj]
Views\MainWindow.axaml.cs(9,9): error CS0103: The name 'InitializeComponent' does not exist in the current context [Template\Template.csproj]
```
I am aware that some of the imports should not be necessary.
Any ideas?

**Comment by timunie** [comment]
did you remember to init git submodules recursively? This is needed to build Avalonia from source.

**Reply by Mrcubix** [reply]
Don't you think i would be getting other error messages if i didn't do? Looking for build steps was the first thing i've done, and found them in Documentation/build.md

**Reply by timunie** [reply]
I don't remember the actual error messages when I faced build issues last time, but re-doing the init helped me. So I shared my thoughts with you and thought it was worth a try. But if you doubt it, ignore my comment.

**Reply by Mrcubix** [reply]
redid the submodule init, nothing changed

**Reply by timunie** [reply, marked as answer]
looking at src/samples/Sandbox there is this switch which may be missing? Can you double-check again?

```diff
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>
+   <IncludeAvaloniaGenerators>true</IncludeAvaloniaGenerators>
  </PropertyGroup>

  <ItemGroup>
    <ProjectReference Include="..\..\src\Avalonia.Diagnostics\Avalonia.Diagnostics.csproj" />
    <ProjectReference Include="..\..\src\Avalonia.Controls.ColorPicker\Avalonia.Controls.ColorPicker.csproj" />
    <ProjectReference Include="..\..\src\Avalonia.Controls.DataGrid\Avalonia.Controls.DataGrid.csproj" />
    <ProjectReference Include="..\..\src\Avalonia.Fonts.Inter\Avalonia.Fonts.Inter.csproj" />
    <ProjectReference Include="..\..\src\Avalonia.Themes.Fluent\Avalonia.Themes.Fluent.csproj" />
  </ItemGroup>
  
  <Import Project="..\..\build\SampleApp.props" />
  <Import Project="..\..\build\ReferenceCoreLibraries.props" />
  <Import Project="..\..\build\BuildTargets.targets" />
  <Import Project="..\..\build\SourceGenerators.props" />
</Project>

```

**Reply by Mrcubix** [reply]
Unfortunately, just this on its own isn't enough, since the path to the generator prop import cannot be used outside of samples, so i had to add the reference myself, it does work now.

At least adding it gave me a clue on how to fix it so i'll mark it as an answer.
